src/resources/subscription.rs

```diff
@@ -129,10 +199,190 @@ impl Subscription {
+    #[serde(skip_serializing_if = "Option::is_none")]
+    customer: Option<CustomerId>,
+
+    /// A cursor for use in pagination.
+    ///
+    /// `ending_before` is an object ID that defines your place in the list.
+    /// For instance, if you make a list request and receive 100 objects, starting with `obj_bar`, your subsequent call can include `ending_before=obj_bar` in order to fetch the previous page of the list.
+    #[serde(skip_serializing_if = "Option::is_none")]
+    ending_before: Option<&'a SubscriptionId>,
+
+    /// Specifies which fields in the response should be expanded.
+    #[serde(skip_serializing_if = "Expand::is_empty")]
+    expand: &'a [&'a str],
+
+    /// A limit on the number of objects to be returned.
+    ///
+    /// Limit can range between 1 and 100, and the default is 10.
+    #[serde(skip_serializing_if = "Option::is_none")]
+    limit: Option<u64>,
+
+    /// The ID of the plan whose subscriptions will be retrieved.
+    #[serde(skip_serializing_if = "Option::is_none")]
+    plan: Option<PlanId>,
+
+    /// A cursor for use in pagination.
+    ///
+    /// `starting_after` is an object ID that defines your place in the list.
+    /// For instance, if you make a list request and receive 100 objects, ending with `obj_foo`, your subsequent call can include `starting_after=obj_foo` in order to fetch the next page of the list.
+    #[serde(skip_serializing_if = "Option::is_none")]
+    starting_after: Option<&'a SubscriptionId>,
+
+    /// The status of the subscriptions to retrieve.
+    ///
+    /// One of: `incomplete`, `incomplete_expired`, `trialing`, `active`, `past_due`, `unpaid`, `canceled`, or `all`.
+    /// Passing in a value of `canceled` will return all canceled subscriptions, including those belonging to deleted customers.
+    /// Passing in a value of `all` will return subscriptions of all statuses.
+    #[serde(skip_serializing_if = "Option::is_none")]
+    status: Option<SubscriptionStatusFilter>,
+}
+
+impl<'a> SubscriptionListParams<'a> {
+    pub fn new() -> Self {
+        SubscriptionListParams {
+            billing: Default::default(),
+            created: Default::default(),
+            current_period_end: Default::default(),
+            current_period_start: Default::default(),
+            customer: Default::default(),
+            ending_before: Default::default(),
+            expand: Default::default(),
+            limit: Default::default(),
+            plan: Default::default(),
+            starting_after: Default::default(),
+            status: Default::default(),
+        }
     }
 }
+
+/// An enum representing the possible values of an `Subscription`'s `billing` field.
+#[derive(Clone, Debug, Deserialize, Serialize)]
+#[serde(rename_all = "snake_case")]
+pub enum SubscriptionBilling {
+    ChargeAutomatically,
+    SendInvoice,
+}
+
+/// An enum representing the possible values of an `Subscription`'s `status` field.
+#[derive(Clone, Debug, Deserialize, Serialize)]
```

Comment:
Should we also derive `PartialEq`, `Eq` for such structures? I think it won't hurt, but allow the users to do conditional checking of status (or any related field), it seems to be more ergonomic in case if the user wants to check one state (writing a `match` instead of one single small if would probably be an overkill)